lhotse/recipes/speechcommands.py

```diff
@@ -0,0 +1,465 @@
+    force_download: bool = False,
+) -> Path:
+    """
+    Download and unzip Speech Commands dataset
+
+    :param speechcommands_version: str, dataset version.
+    :param target_dir: Pathlike, the path of the dir to store the dataset.
+    :param force_download: bool, if True, download the archive even if it already exists.
+
+    :return: the path to downloaded and extracted directory with data.
+    """
+    target_dir = Path(target_dir)
+    corpus_dir = target_dir / f"SpeechCommands{speechcommands_version}"
+    corpus_dir.mkdir(parents=True, exist_ok=True)
+
+    dataset_parts = (
+        f"speech_commands_v0.0{speechcommands_version}",
+        f"speech_commands_test_set_v0.0{speechcommands_version}",
+    )
+
+    for part in tqdm(
+        dataset_parts, desc=f"Downloading Speech Commands v0.0{speechcommands_version}"
+    ):
+        logging.info(f"Processing split: {part}")
+        part_dir = corpus_dir / part
+        completed_detector = part_dir / ".completed"
+        if completed_detector.is_file():
+            logging.info(f"Skipping {part} because {completed_detector} exists.")
+            continue
+        # Process the archive.
+        tar_name = f"{part}.tar.gz"
+        tar_path = corpus_dir / tar_name
+        if force_download or not tar_path.is_file():
+            urlretrieve_progress(
+                f"http://download.tensorflow.org/data/{tar_name}",
+                filename=tar_path,
+                desc=f"Downloading {tar_name}",
+            )
+        # Remove partial unpacked files, if any, and unpack everything.
+        shutil.rmtree(part_dir, ignore_errors=True)
+        with tarfile.open(tar_path) as tar:
+            tar.extractall(part_dir)
+        completed_detector.touch()
+
+    return corpus_dir
+
+
+def download_speechcommands1(
+    target_dir: Pathlike = ".",
+    force_download: bool = False,
+) -> Path:
+    """
+    Download and unzip the Speech Commands v0.01 data.
+    :param target_dir: Pathlike, the path of the dir to store the dataset.
+    :param force_download: bool, if True, download the archive even if it already exists.
+    :return: the path to downloaded and extracted directory with data.
+    """
+
+    return _download_speechcommands(
+        speechcommands_version="1",
+        target_dir=target_dir,
+        force_download=force_download,
+    )
+
+
+def download_speechcommands2(
+    target_dir: Pathlike = ".",
+    force_download: bool = False,
+) -> Path:
+    """
+    Download and unzip the Speech Commands v0.02 data.
+    :param target_dir: Pathlike, the path of the dir to store the dataset.
+    :param force_download: bool, if True, download the archive even if it already exists.
+    :return: the path to downloaded and extracted directory with data.
+    """
+
+    return _download_speechcommands(
+        speechcommands_version="2",
+        target_dir=target_dir,
+        force_download=force_download,
+    )
+
+
+def _prepare_train_valid(
```

Comment:
could you split this function into separate prepare_train and prepare_valid? I don't think you really need a generator here if you restructure the code, and the way to use this as-is is a little confusing.

Reply:
OK, I will split it.